app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -66,12 +70,16 @@ class OtherUserProfileScreenViewModel @Inject constructor(
     private val userTypeMapper: UserTypeMapper,
     private val wireSessionImageLoader: WireSessionImageLoader,
     private val observeConversationRoleForUser: ObserveConversationRoleForUserUseCase,
-    qualifiedIdMapper: QualifiedIdMapper,
-    private val updateMemberRole: UpdateConversationMemberRoleUseCase
+    private val updateMemberRole: UpdateConversationMemberRoleUseCase,
+    private val removeMemberFromConversation: RemoveMemberFromConversationUseCase,
+    qualifiedIdMapper: QualifiedIdMapper
 ) : ViewModel() {
 
     var state: OtherUserProfileState by mutableStateOf(OtherUserProfileState())
 
+    var removeConversationMemberDialogState: DialogState<PreservedState<RemoveConversationMemberState>>
```

Comment:
I'm not sure about the idea of passing the `RemoveConversationMemberState` through all the "preserved" states, we could separate both, for instance `dialogState` and `dialogActionState`, or it could be done the other way around by just adding a parameter to `RemoveConversationMemberState`, since we don't use `PreservedState.Error` state anyway so it can be even:
```
data class RemoveConversationMemberState(
    ...
    isLoading: Boolean = false
)
```
But maybe it's just me thinking that it's a too big gun for the job 😄